### PR TITLE
Add previous command

### PR DESCRIPTION
**Pull request by dzove855**
Add command to switch to the previous channel

### CODE REVIEW

File: birch [hunk 1]
```diff
@@ -183,6 +183,12 @@ cmd() {
             ln -sf "$chan" .c
             kill -28 0 
         ;;
+        
+        '/previous'*)
```
Comment (dylanaraps) [inline birch]: 'prev' would be better.

Comment (dylanaraps) [reply birch]: This is probably the best way:

```sh
        '/next'*) 
            chan=${cl[z = z + 1 >= ${#cl[@]} ? 0 : z + 1]} 
        ;;&

        '/prev'*) 
            chan=${cl[z = z - 1 < 0 ? ${#cl[@]} : z - 1]}
        ;;&

        '/next'*|'/prev'*)
            ln -sf "$chan" .c
            kill -28 0 
        ;;
```

Comment (dzove855) [reply birch]: It might be better like this. I will test it. Actually i had done all in on case with an internal case in case but it was horrible..

Comment (dzove855) [reply birch]: Implmented it like this. Didn't know about ;;& . Thanks for the suggestion 👍

Comment (dylanaraps) [reply birch]: `;&` will fall through regardless of checking the next pattern.

`;;&` will fall through and check the patterns.

File: birch [hunk 2]
```diff
@@ -178,10 +178,17 @@ cmd() {
...
          ;;
 
-        '/next'*) 
-            chan=${cl[z = z + 1 >= ${#cl[@]} ? 0 : z + 1]} 
+        '/next'*)
+            chan=${cl[z = z + 1 >= ${#cl[@]} ? 0 : z + 1]}
+        ;;&
+
+        '/prev'*)
+            chan=${cl[z = z - 1 < 0 ? ${#cl[@]} : z - 1]}
```
Comment (dylanaraps) [inline birch]: There's an off-by-one error here. Try `<= 0`?

Comment (dylanaraps) [reply birch]: This works:

```
            chan=${cl[z = z - 1 < 0 ? ${#cl[@]}-1 : z - 1]}
```

Comment (dzove855) [reply birch]: Thanks. Commited.

**Comment by dylanaraps** [comment]
:+1:

**Comment by LoLei** [comment]
Nice I was just thinking about adding this myself!

**Pull request merged**